pantheon-files-daemon/FileManager1.vala

```diff
@@ -43,7 +43,8 @@ public class FileManager1 : Object {
         string cmd = "pantheon-files -t";
 
         foreach (string s in uris) {
-            cmd += (" " + s);
+            s = s.replace ("'", "%27").replace ("%", "%%");
+            cmd += (" \'" + PF.FileUtils.sanitize_path (s, null, false) + "\'");
```

Comment:
Ah, your code looks more robust and doesn't mess with the existing sanitize method. I didn't realise there needed to be a null check because the return type of the existing sanitize method is `string` and not `string?`. I think we can just close this PR once your PR gets merged to master.